### PR TITLE
fix(test) : update UI tests, typescript version, package.json

### PR DESCRIPTION
**Pull request by Raunak1203**
<strong>Note: If there are pending changes to the PR, prefix the PR title with "WIP" and add the label "DO NOT MERGE"</strong>

### Mandatory
 - [x] What does this PR do? 
-This PR updates UI tests with typescript version 2.9.2 and pack the page objects to reuse with Planner- Platform Integration tests in fabric8-test 
- [x] What issue/task does this PR references?
https://openshift.io/openshiftio/Openshift_io/plan/detail/758
- [ ] Are the tests Included?
___
### Optional
- [ ] Is the documentation Included?

- [ ] Are the Release Notes included?

- [ ] @mention(s) to expected reviewer(s) included

### CODE REVIEW

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

File: src/tests/page_objects/base.page.ts [hunk 1]
```diff
@@ -12,8 +11,8 @@ export abstract class BasePage {
   // add logging mixin
 
   name: string = '...';
-  log: (action: string, ...msg: string[]) => void;
-  debug: (context: string, ...msg: string[]) => void;
+  log!: (action: string, ...msg: string[]) => void;
```
[the remark this answers is not in the document]
Comment (Raunak1203) [reply src/tests/page_objects/base.page.ts]: @jarifibrahim to fix compilation error - Property '' has no initializer and is not definitely assigned in the constructor, strictPropertyInitialization compiler option is by default on. To fix the compilation error either this "strictPropertyInitialization" should be false in tsConfig or by initializing the properties by appending a ! to them

File: src/tests/page_objects/base.page.ts [hunk 2]
```diff
@@ -12,8 +11,8 @@ export abstract class BasePage {
...
+  debug!: (context: string, ...msg: string[]) => void;
```
[the remark this answers is not in the document]
Comment (Raunak1203) [reply src/tests/page_objects/base.page.ts]: same reason as above

File: cico_test_prod.sh [hunk 3]
```diff
@@ -50,7 +50,7 @@ CID=$(docker run --detach=true \
 # Get all the deps in
 echo "Container name: $CID"
 
-docker exec $CID bash -c 'cd tests'
+docker exec $CID bash -c 'npm i && cd tests'
```
[the remark this answers is not in the document]
Comment (jarifibrahim) [reply cico_test_prod.sh]: I spoke to @Raunak1203 and it looks like we need this.

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Comment by Raunak1203** [comment]
[test]

**Pull request merged**